recipe/meta.yaml

```diff
@@ -30,7 +31,9 @@ requirements:
   build:
     - {{ compiler("c") }}
     - {{ compiler("cxx") }}
+    - sysroot_{{ target_platform }} 2.17    # [linux]
```

Comment:
Interestingly was able to get the package to build locally with only this change (adding the `sysroot` to `requirements/build`)

Given this, wonder if we can drop the `patchelf` bits

Reply:
Not sure how you tested it, this is not sufficient when running `build-locally.py`

Reply:
Did the following locally

```
docker run --rm -it quay.io/condaforge/linux-anvil-aarch64

cat >>~/.condarc <<EOF
conda-build:
  error_overlinking: true
EOF

git clone https://github.com/adibbley/cuda-sanitizer-api-feedstock -b cuda-12.3.2
cd cuda-sanitizer-api-feedstock
git reset --hard 25c14ad89edd769dfe5e4955f7d44f5313b91926

git apply -
diff --git a/recipe/meta.yaml b/recipe/meta.yaml
index 71d935e..9a0d82b 100644
--- a/recipe/meta.yaml
+++ b/recipe/meta.yaml
@@ -30,6 +30,7 @@ requirements:
   build:
     - {{ compiler("c") }}
     - {{ compiler("cxx") }}
+    - sysroot_{{ target_platform }} 2.17    # [linux]
     - arm-variant * {{ arm_variant_type }}  # [aarch64]
   host:
     - cuda-version {{ cuda_version }}
<ctrl + d>

conda build -m .ci_support/linux_aarch64_.yaml recipe/
```

Reply:
Looks like you tested aarch64, this error only occurred on x86_64 AFAICT. First CI from before any changes: https://github.com/conda-forge/cuda-sanitizer-api-feedstock/pull/5/checks?check_run_id=20821915045

Reply:
Oops thanks for catching that. Am on an ARM machine where this is a little hard to test

Let's stick with what we have already. If we come up with a better solution, we can always do another PR